tests/integration/echo/main_test.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package echo
+
+import (
+	"testing"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/galley"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/pilot"
+	"istio.io/istio/pkg/test/framework/resource"
+)
+
+var (
+	ist istio.Instance
+	p   pilot.Instance
+	g   galley.Instance
+)
+
+func TestMain(m *testing.M) {
+	framework.
+		NewSuite("echo_test", m).
+		SetupOnEnv(environment.Kube, istio.Setup(&ist, nil)).
+		Setup(func(ctx resource.Context) (err error) {
```

Comment:
Is there a reason to hoisting pilot and galley instances like this?

Reply:
It just gets galley and pilot out of the test code.  We need a main anyway, treating them as globals just seems cleaner. Also if we had multiple TestXXX methods, they wouldn't need to duplicate the same boilerplate for pilot/galley.